app/components/administrative-body-select.js

```diff
@@ -21,6 +21,17 @@ const VALID_ADMINISTRATIVE_BODY_CLASSIFICATIONS = [
   "http://data.vlaanderen.be/id/concept/BestuursorgaanClassificatieCode/5ab0e9b8a3b2ca7c5e000008" //	"Vast Bureau"
 ];
 
+/**
+ * @typedef {Object} Args
+ *
+ * @property {string} id Input id so it can be linked with a label
```

Comment:
What does `@property` do exactly? What does TS do with this annotation?

Reply:
This whole typedef is just the syntax for defining an object in jsdoc, as seen [here](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#typedef-callback-and-param)

We then document the component by saying it extends `Component<Args>` which is how glimmer components are typed. In theory, with a compliant typescript service, this should allow your editor to provide type hinting and autocompletion on the `this.args` object. In practice, this works quite well in emacs and vscode, not so great in webstorm. 
You could even setup the ts compiler to really typecheck these and error out when it's not right, but given we have such little type information in the project I think that's not a great idea.

So its usefulness for ts is a bit of a mixed bag. But since we need to document the arguments for a component _anyway_ (cause you know, thats the interface of the thing) I figured why not use the syntax that can actually be used by tooling.

You see im not going all the way here by not importing the `BestuursOrgaan` model type, cause that becomes cumbersome very quickly (the syntax is hideous) and provides no extra benefit since our models are also not typed/documented.

Reply:
Ah, thanks for the info! The "property" part made me think it was related to the class somehow. I'm not familiar with adding types through comments, as evidenced :smile:. Did you take a look at the [glint](https://www.salsify.com/blog/engineering/introducing-glint-a-typed-template-solution-for-glimmerx-and-ember) project btw? If not I think you will like it :grimacing:.

Reply:
No, not really, no. But it's cool that something like that is in the works.